components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/internal/service/impl/UserAccountRecoveryManager.java

```diff
@@ -312,10 +312,7 @@ public String getUsernameByClaims(Map<String, String> claims, String tenantDomai
                 resultedUserList.addAll(abstractUserStoreManager.getUserListWithID(operationalCondition, domain,
                         UserCoreConstants.DEFAULT_PROFILE, 2, 1, null, null));
                 if (resultedUserList.size() > 1) {
-                    if (log.isDebugEnabled()) {
-                        log.debug("Multiple users matched for given claims set : " +
-                                Arrays.toString(resultedUserList.toArray()));
-                    }
+                    log.warn("Multiple users matched for the given claims set.");
```

Comment:
what if we want to check the user list? can't we keep both the debug and warn log?

Reply:
output of the existing debug log is as follows.
```
[2023-07-21 09:34:00,943] [bda4e1e8-e14c-4f34-93af-05d5f3eeee16]  DEBUG {org.wso2.carbon.identity.recovery.internal.service.impl.UserAccountRecoveryManager} - Multiple users matched for given claims set : [org.wso2.carbon.user.core.common.User@d722e87b, org.wso2.carbon.user.core.common.User@b900b21a]

```
user list is not readable in the existing debug log, and also I don't think it is good to expose PIIs(username, email addresses) through logs.